server/src-lib/Hasura/RQL/Types/BoolExp.hs

```diff
@@ -307,22 +310,23 @@ opExpToJPair f = \case
   where
     opExpsToJSON = object . map (opExpToJPair f)
 
-data AnnBoolExpFld a
-  = AVCol !PGColumnInfo ![OpExpG a]
-  | AVRel !RelInfo !(AnnBoolExp a)
-  deriving (Show, Eq, Functor, Foldable, Traversable, Generic)
-instance (NFData a) => NFData (AnnBoolExpFld a)
-instance (Cacheable a) => Cacheable (AnnBoolExpFld a)
-instance (Hashable a) => Hashable (AnnBoolExpFld a)
+data AnnBoolExpFld (b :: Backend) a
+  = AVCol !(ColumnInfo b) ![OpExpG a]
```

Comment:
TODO: `OpExpG a -> OpExpG b a`

Reply:
For now I added `XAILIKE` and `XANILIKE` type families that are part of the corresponding constructors. Indeed it may also make sense to have an extension constructor, but for the sake of scope creep let's add that when we have a good use case for it.